src/js/apps/clinicians/clinicians-all_app.js

```diff
@@ -18,6 +18,9 @@ export default SubRouterApp.extend({
   viewEvents: {
     'click:addClinician': 'onClickAddClinician',
   },
+  onBeforeStop() {
+    this.clinicians = null;
+  },
```

Comment:
I believe this issue was caused by `this.clinicians` being cached when a user visits the clinicians page more than once.

As a result, the `Find in list...` input would always have `state.isDisabled = true` when the page is loaded after the first visit. Which means it would never show as disabled.

Another solution I came up with was to update the `showSearchView()` function to this:

```js
showSearchView(setAsDisabled = false) {
  const isDisabled = setAsDisabled || !this.clinicians || !this.clinicians.length;

  const searchComponent = this.showChildView('search', new SearchComponent({
    state: {
      query: this.getState('searchQuery'),
      isDisabled,
    },
  }));

  this.listenTo(searchComponent.getState(), 'change:query', this.setSearchState);
},
```

And then do `this.showSearchView(true)` in the `onBeforeStart()`.

But clearing `this.clinicians` in the `onBeforeStop()` seemed more simple.